lib/propolis/src/block/mem_async.rs

```diff
@@ -59,17 +45,19 @@ impl MemAsyncBackend {
             ));
         }
 
-        let seg = MmapSeg::new(size)?;
+        let seg = MmapSeg::new(size as usize)?;
 
         Ok(Arc::new(Self {
             seg: Arc::new(seg),
 
             workers,
             scheduler: block::Scheduler::new(),
 
-            read_only,
-            block_size,
-            sectors: size / block_size,
+            info: block::DeviceInfo {
+                block_size,
+                total_size: size / block_size as u64,
+                writable: opts.read_only.unwrap_or(false),
```

Comment:
```suggestion
                writable: !opts.read_only.unwrap_or(false),
```

Mind updating `DeviceInfo::writable` to `read_only` while you're here to be consistent everywhere

Reply:
> Mind updating `DeviceInfo::writable` to `read_only`

Should be done.